tests/backends/data/test_es.py

```diff
@@ -0,0 +1,622 @@
+    # pylint: disable=invalid-name,unused-argument
+    backend = es_backend()
+    # Insert documents
+    bulk(
+        backend.client,
+        (
+            {"_index": ES_TEST_INDEX, "_id": idx, "_source": {"id": idx}}
+            for idx in range(10)
+        ),
+    )
+    # As we bulk insert documents, the index needs to be refreshed before making
+    # queries.
+    backend.client.indices.refresh(index=ES_TEST_INDEX)
+    expected = [{"id": idx} for idx in range(10)]
+    result = backend.read(raw_output=True)
+    for i, document in enumerate(result):
+        assert isinstance(document, bytes)
+        assert json.loads(document).get("_source") == expected[i]
+
+
+def test_backends_data_es_data_backend_read_method_without_raw_ouput(es, es_backend):
+    """Tests the `ESDataBackend.read` method with `raw_output` set to `False`."""
+    # pylint: disable=invalid-name,unused-argument
+    backend = es_backend()
+    # Insert documents
+    bulk(
+        backend.client,
+        (
+            {"_index": ES_TEST_INDEX, "_id": idx, "_source": {"id": idx}}
+            for idx in range(10)
+        ),
+    )
+    # As we bulk insert documents, the index needs to be refreshed before making
+    # queries.
+    backend.client.indices.refresh(index=ES_TEST_INDEX)
+    expected = [{"id": idx} for idx in range(10)]
+    result = backend.read()
+    for i, document in enumerate(result):
+        assert isinstance(document, dict)
+        assert document.get("_source") == expected[i]
+
+
+def test_backends_data_es_data_backend_read_method_with_query(es, es_backend):
+    """Tests the `ESDataBackend.read` method with a query."""
+    # pylint: disable=invalid-name,unused-argument
+    # Insert documents
+    backend = es_backend()
+    bulk(
+        backend.client,
+        (
+            {
+                "_index": ES_TEST_INDEX,
+                "_id": idx,
+                "_source": {"id": idx, "modulo": idx % 2},
+            }
+            for idx in range(10)
+        ),
+    )
+    # As we bulk insert documents, the index needs to be refreshed before making
+    # queries.
+    backend.client.indices.refresh(index=ES_TEST_INDEX)
+    # Find every even item.
+    query = ESQuery(query={"query": {"term": {"modulo": 0}}})
+    results = list(backend.read(query=query))
+    assert len(results) == 5
+    assert results[0]["_source"]["id"] == 0
+    assert results[1]["_source"]["id"] == 2
+    assert results[2]["_source"]["id"] == 4
+    assert results[3]["_source"]["id"] == 6
+    assert results[4]["_source"]["id"] == 8
+
+    # Find every odd item.
+    query = {"query": {"query": {"term": {"modulo": 1}}}}
+    results = list(backend.read(query=query))
+    assert len(results) == 5
+    assert results[0]["_source"]["id"] == 1
+    assert results[1]["_source"]["id"] == 3
+    assert results[2]["_source"]["id"] == 5
+    assert results[3]["_source"]["id"] == 7
+    assert results[4]["_source"]["id"] == 9
```

Comment:
Maybe only go up to 5 statements and remove a few lines of code ? ☁️

Reply:
Good Idea) Thanks) Updated this part)